File: RepoTransBench/target_projects/Python/C++/JoshData_pdf-redactor/public_tests/test_public_redactor.cpp
#include <gtest/gtest.h>
#include "pdf_redactor.hpp"
#include <sstream>
#include <vector>
#include <regex>
#include <string>

TEST(PublicRedactorTest, BasicRedaction) {
    std::string pdf_str = "%PDF-1.4\n% Public test: secret12345 replaced\nxyz 654-32-1987 zyx\n%%EOF";
    std::stringstream pdf_in(pdf_str);
    pdf_redactor::RedactorOptions options;
    options.content_filters = {
        {std::regex("\\b654-32-1987\\b"), [](const std::smatch&) { return std::string("[REDACTED-ID]"); }}
    };
    std::stringstream pdf_out;
    pdf_redactor::redactor(options, &pdf_in, &pdf_out);
    EXPECT_NE(pdf_out.str().find("[REDACTED-ID]"), std::string::npos);
}

TEST(PublicRedactorTest, UnicodeFilter) {
    std::string pdf_str = "%PDF-1.4\nUnusual symbol: §\nID 88-99-7766\n%%EOF";
    std::stringstream pdf_in(pdf_str);
    pdf_redactor::RedactorOptions options;
    options.content_filters = {
        {std::regex("\\b88-99-7766\\b"), [](const std::smatch&) { return std::string("<REMOVED>"); }}
    };
    std::stringstream out;
    pdf_redactor::redactor(options, &pdf_in, &out);
    EXPECT_NE(out.str().find("<REMOVED>"), std::string::npos);
}

TEST(PublicRedactorTest, MultilineFilter) {
    std::string pdf_str = "%PDF-1.4\nFirstLine\nID: 222-33-4444\nSecondLine\n%%EOF";
    std::stringstream pdf_in(pdf_str);
    pdf_redactor::RedactorOptions options;
    options.content_filters = {
        {std::regex("222-33-4444"), [](const std::smatch&) { return std::string("*****"); }}
    };
    std::stringstream out;
    pdf_redactor::redactor(options, &pdf_in, &out);
    EXPECT_NE(out.str().find("*****"), std::string::npos);
}

File: RepoTransBench/target_projects/Python/Rust/JoshData_pdf-redactor/public_tests/test_public_redactor.rs
use pdf_redactor_rs::pdf_redactor::{RedactorOptions, redactor};
use regex::Regex;

#[test]
fn test_basic_redaction() {
    let mut options = RedactorOptions::default();
    options.content_filters = vec![
        (
            Regex::new(r"\b654-32-1987\b").unwrap(),
            Box::new(|_m: &str| "[REDACTED-ID]".to_string()),
        ),
    ];
    let pdf_in = b"%PDF-1.4\n% Public test: secret12345 replaced\nxyz 654-32-1987 zyx\n%%EOF".to_vec();
    let mut out = vec![];
    redactor(&mut options, Some(pdf_in.clone()), &mut out).unwrap();
    assert!(out.contains(&b'['));
    let s = String::from_utf8_lossy(&out);
    assert!(s.contains("[REDACTED-ID]"));
}

#[test]
fn test_unicode_filter() {
    let mut options = RedactorOptions::default();
    options.content_filters = vec![(
        Regex::new(r"\b88-99-7766\b").unwrap(),
        Box::new(|_m: &str| "<REMOVED>".to_string()),
    )];
    let pdf_in = "%PDF-1.4\nUnusual symbol: §\nID 88-99-7766\n%%EOF".as_bytes().to_vec();
    let mut out = vec![];
    redactor(&mut options, Some(pdf_in.clone()), &mut out).unwrap();
    let s = String::from_utf8_lossy(&out);
    assert!(s.contains("<REMOVED>"));
}

#[test]
fn test_multiline_filter() {
    let mut options = RedactorOptions::default();
    options.content_filters = vec![(
        Regex::new(r"222-33-4444").unwrap(),
        Box::new(|_m: &str| "*****".to_string()),
    )];
    let pdf_in = b"%PDF-1.4\nFirstLine\nID: 222-33-4444\nSecondLine\n%%EOF".to_vec();
    let mut out = vec![];
    redactor(&mut options, Some(pdf_in.clone()), &mut out).unwrap();
    let s = String::from_utf8_lossy(&out);
    assert!(s.contains("*****"));
}